src/ClientData/FunctionUtils.cpp

```diff
@@ -44,12 +45,23 @@ uint64_t Offset(const FunctionInfo& func, const ModuleData& module) {
 
 std::optional<uint64_t> GetAbsoluteAddress(const orbit_client_protos::FunctionInfo& func,
                                            const ProcessData& process, const ModuleData& module) {
-  std::optional<uint64_t> base_address = process.GetModuleBaseAddress(func.module_path());
-  if (!base_address.has_value()) {
+  std::vector<uint64_t> base_address =
+      process.GetModuleBaseAddresses(module.file_path(), module.build_id());
+  if (base_address.empty()) {
     return std::nullopt;
   }
 
-  return func.address() + base_address.value() - module.load_bias();
+  if (base_address.size() > 1) {
+    ERROR(
+        "Found multiple mappings for \"%s\" with build_id=%s [%s]: "
+        "will use the first one as a base address",
+        module.file_path(), module.build_id(),
+        absl::StrJoin(base_address, ",", [](std::string* out, uint64_t address) {
+          return out->append(absl::StrFormat("%#x", address));
+        }));
+  }
+
```

Comment:
Nit: Maybe add a CHECK(base_address.size() == 0) here.

Reply:
we have an early return in line 50

Reply:
I will add it anyways.

Reply:
Florian, you mean `== 1`?

Reply:
For sure, I meant `== 1`, sorry. Also, yes that is not needed, as we do have the early two early returns, and probably the code will not evolve too much, but I feel a bit saver with the check (e.g. if we add some code after the early return).

Reply:
done.